### PR TITLE
Migrate FxCop Legacy (Legacy analysis) to FxCop Analyzers (Source analysis)

### PR DESCRIPTION
**Pull request by KenitoInc**
- [X] Installed `Microsoft.CodeAnalysis.FxCopAnalyzers v3.0.0`
- [X] Changes some severity levels from Error to warning (To prevent build from failing. They will however be reverted to Errors once all warnings have been fixed)
- [X] Ported the `ruleset` file to `.editorconfig` file.

Like legacy analysis rule violations, source code analysis violations appear in the Error List window in Visual Studio. In addition, source code analysis violations also show up in the code editor as `squiggles` under the offending code. This ensures that the developer will get notified of any violations live when coding.

### CODE REVIEW

File: tools/CustomMSBuild/ODataRuleSets.ruleset [hunk 1]
```diff
@@ -115,22 +115,22 @@
...
-    <Rule Id="CA1712" Action="Error" />
+    <Rule Id="CA1712" Action="Warning" />
     <Rule Id="CA1713" Action="Error" />
     <Rule Id="CA1714" Action="Error" />
     <Rule Id="CA1715" Action="Error" />
     <Rule Id="CA1716" Action="Error" />
     <Rule Id="CA1717" Action="Error" />
     <Rule Id="CA1719" Action="Error" />
-    <Rule Id="CA1720" Action="Error" />
+    <Rule Id="CA1720" Action="Info" />
```
Comment (g2mula) [inline tools/CustomMSBuild/ODataRuleSets.ruleset:125]: Why not warning?

File: tools/CustomMSBuild/ODataRuleSets.ruleset [hunk 2]
```diff
@@ -596,4 +596,25 @@
     <Rule Id="C6995" Action="None" />
     <Rule Id="C6997" Action="None" />
   </Rules>
+  <Rules AnalyzerId="Microsoft.CodeQuality.Analyzers" RuleNamespace="Microsoft.CodeQuality.Analyzers">
+    <Rule Id="CA1720" Action="Info" />
```
Comment (g2mula) [inline tools/CustomMSBuild/ODataRuleSets.ruleset:600]: Why not warning?

File: src/Microsoft.OData.Client/ALinq/TypeSystem.cs [hunk 3]
```diff
@@ -16,9 +16,6 @@ namespace Microsoft.OData.Client
     /// <summary>Utility functions for processing Expression trees</summary>
     internal static class TypeSystem
     {
-        /// <summary>Used for comparison with external assemblies for silverlight like Microsoft.VisualBasic.</summary>
-        private const string OfficialSilverLightPublicKeyToken = "31bf3856ad364e35";
-
```
Comment (g2mula) [inline src/Microsoft.OData.Client/ALinq/TypeSystem.cs:21]: Yipppeee!

File: tools/CustomMSBuild/ODataRuleSets.ruleset (hunk 1 above)
Comment (KenitoInc) [reply tools/CustomMSBuild/ODataRuleSets.ruleset:125]: Some of these rules were changed to `Info` when we upgraded to FxCop analyzers. Let me manually adjust them to `Warnings`

Comment (KenitoInc) [reply tools/CustomMSBuild/ODataRuleSets.ruleset:125]: For CA1720: Identifiers should not contain type names.
However, in EdmLib, we have Enum properties with type name in `EdmPrimitiveTypeKind`

**Comment by KenitoInc** [comment]
> Would you please help on Core.Edm.Spatial? :)

@xuzhg Do you mean adding FxCop analyzers for Core, Edm and Spatial? I added FxCop analyzers to all the four projects

**Comment by StevenLiekens** [comment]
@KenitoInc there's a leftover reference to the rulesets file and it causes warnings (MSB3884). Is it safe to remove?

https://github.com/OData/odata.net/blob/39aad84e04fcedbda5ce00fd98ed5ad4f62d7025/tools/CustomMSBuild/Build.props#L136-L139

**Pull request merged**